### PR TITLE
Remove unnecessary use of `sudo pip`.

### PR DESCRIPTION
**Pull request by jwhui**

### CODE REVIEW

File: .travis/before_install.sh [hunk 1]
```diff
@@ -39,21 +39,15 @@ cd /tmp || die
 [ $TRAVIS_OS_NAME != linux ] || {    
     sudo apt-get update || die
 
-    sudo -H pip install --upgrade pip || die
```
Comment (bukepo) [inline .travis/before_install.sh]: Since my pip is already the latest, I don't have an easy way to try it. Just wondering, does the next line need sudo?

Comment (jwhui) [inline .travis/before_install.sh]: sudo should not be necessary if everything is being done via the same user.

File: .travis/before_install.sh [hunk 2]
```diff
@@ -39,21 +39,15 @@ cd /tmp || die
...
     pip install --upgrade pip || die
```
Comment (bukepo) [inline .travis/before_install.sh]: This line seems to upgrade the system wide `pip`. Does this require sudo privilege?

Comment (jwhui) [inline .travis/before_install.sh]: In general, use of `sudo` with pip is discouraged.  Also, Travis seems to indicate that `sudo` is not required for the test scripts.

**Pull request merged**